surrealdb/surrealdb_test.go

```diff
@@ -0,0 +1,220 @@
+package surrealdb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/surrealdb"
+	"os"
+	"testing"
+	"time"
+)
+
+var (
+	// surrealDb is the default image used for running surrealdb in tests.
+	surrealDb                   = "surrealdb/surrealdb:latest"
+	surrealDbImageEnvVar string = "TEST_SURREALDB_IMAGE"
+	surrealDbUser        string = "root"
+	surrealDbPass        string = "root"
+)
+
+func newTestStore(t testing.TB) (*Storage, error) {
+	t.Helper()
+
+	img := surrealDb
+	if imgFromEnv := os.Getenv(surrealDbImageEnvVar); imgFromEnv != "" {
+		img = imgFromEnv
+	}
+	surrealdbContainer, err := surrealdb.Run(context.Background(), img)
+	if err != nil {
+		return nil, err
+	}
+
+	testcontainers.CleanupContainer(t, surrealdbContainer)
+
+	host, err := surrealdbContainer.Host(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	port, err := surrealdbContainer.MappedPort(context.Background(), "8000")
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf("ws://%s:%s", host, port.Port())
```

Comment:
suggestion: simplify using module's API
```suggestion
	url, err := surrealdbContainer.URL(ctx)
	if err != nil {
		return nil, err
	}
```